internal/hcp/internal/types/telemetry_state.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package types
+
+import (
+	"errors"
+
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/hashicorp/consul/internal/resource"
+	pbhcp "github.com/hashicorp/consul/proto-public/pbhcp/v2"
+)
+
+type DecodedTelemetryState = resource.DecodedResource[*pbhcp.TelemetryState]
+
+var (
+	telemetryStateConfigurationNameError = errors.New("only a single Telemetry resource is allowed and it must be named global")
+)
+
+func RegisterTelemetryState(r resource.Registry) {
+	r.Register(resource.Registration{
+		Type:     pbhcp.TelemetryStateType,
+		Proto:    &pbhcp.TelemetryState{},
+		Scope:    resource.ScopeCluster,
+		Validate: ValidateTelemetryState,
+	})
+}
+
+var ValidateTelemetryState = resource.DecodeAndValidate(validateTelemetryState)
+
+func validateTelemetryState(res *DecodedTelemetryState) error {
```

Comment:
Maybe worth also validating here that `include_list` is set or at least not nil. At the time of writing, that being empty means we'll get no metrics. And presumably it being empty means something went wrong somewhere. I can see us _maybe_ setting that to an empty list in an effort to turn off metrics pushing during an incident, but it shouldn't be nil

Same for `labels` (which is the only way to can tell clusters apart within an org) and `endpoint` (which we don't have a fallback for right now)